### PR TITLE
[MC-1886] Support Docker image

### PR DESCRIPTION
**Pull request by wisrob**

### CODE REVIEW

**Comment by wisrob** [comment]
Related PR with the use of both the new module version. Both `Image` and `Zip` types are used.

https://github.com/craft-machine/risk-service/pull/128

File: aws_lambda_function.tf [hunk 1]
```diff
@@ -1,85 +1,52 @@
...
-
-  tracing_config {
-    mode = "PassThrough"
-  }
-
-  timeout = var.timeout
-  runtime = var.runtime
-  tags    = var.tags
+  function_name = var.name
+  handler       = local.handler_name
```
Comment (adys) [inline aws_lambda_function.tf:13]: @wisrob could we get this from var.handler_name instead? I think we can just set "lambda_function.lambda_function as the default value of handler_name var.

We don't need [this](https://github.com/craft-machine/terraform-aws-lambda-function/pull/2/files#diff-3c3e6c55da96743d977302f9e9f8316b9d388c2efee6621b45b8d39d084b8f78R2) I think.

File: variables.tf [hunk 2]
```diff
@@ -79,177 +90,244 @@ variable "layers" {
   default     = []
 }
 
+variable "package_type" {
+  description = "The Lambda deployment package type. Valid options: Zip or Image"
+  type        = string
+  default     = "Zip"
```
Comment (adys) [inline variables.tf:96]: @wisrob I know you are using this module to deploy a function with docker image. However, have you tested the "Zip" package type?

File: aws_lambda_function.tf [hunk 3]
```diff
@@ -1,85 +1,52 @@
...
-  sqs_stream_id_env       = tomap({"sqs_stream_id": local.output_sqs_stream_id})
+  kinesis_stream_name_env = tomap({ "kinesis_stream_name" : local.output_kinesis_stream_name })
+  sns_stream_name_env     = tomap({ "sns_stream_name" : local.output_sns_stream_name })
+  sqs_stream_id_env       = tomap({ "sqs_stream_id" : local.output_sqs_stream_id })
 }
 
-resource "aws_lambda_function" "aws_lambda_function" {
-  count = var.vpc_enabled ? 0 : 1
+module "lambda_function" {
+  source  = "terraform-aws-modules/lambda/aws"
```
Comment (adys) [inline aws_lambda_function.tf:9]: @wisrob could you add a simple note to the README file of this module saying "Python 3.6 or newer is required to use this module".

It's a new requirement after we've introduced https://github.com/terraform-aws-modules/terraform-aws-lambda.

Comment (wisrob) [reply aws_lambda_function.tf:8]: Added https://github.com/craft-machine/terraform-aws-lambda-function/blob/97ea2bfd9b84e93dfd96977541722c71d0f69da3/README.md

**Review by adys — APPROVED** [review]

**Pull request merged**